doc/releases/release-notes-4.0.rst

```diff
@@ -1381,6 +1382,7 @@ Libraries / Subsystems
   * :ref:`_POSIX_THREAD_SAFE_FUNCTIONS <posix_option_thread_safe_functions>` improvements:
 
     * Support for :c:func:`asctime_r`, :c:func:`ctime_r`, and :c:func:`localtime_r`.
+    * Added stub for :c:func:`getpwnam_r`, :c:func:`getpwuid_r`, :c:func:`getgrgid_r` and :c:func:`getgrnam_r`.
```

Comment:
I think this is maybe Ok for a backport PR to v4.0. 

However, looking into things a bit deeper, `getpwnam_r()`, `getpwuid_r()`, `getgrgid_r()`, and `getgrnam_r()` belong to a separate Option Group - 
`POSIX_SYSTEM_DATABASE_R`. And the non-reentrant versions of these belong to the `POSIX_SYSTEM_DATABASE` Option Group, and `_POSIX_THREAD_SAFE_FUNCTIONS` is more of an Option / Feature Test Macro.

So really what we need here is a new `Kconfig.system_database` file with two new Kconfig options (same as the option group names).

`_POSIX_THREAD_SAFE_FUNCTIONS` should probably be a hidden Kconfig option that depends on other Kconfig values being enabled. Namely, `CONFIG_POSIX_SYSTEM_DATABASE_R`, `CONFIG_POSIX_C_LANG_SUPPORT_R`, `CONFIG_POSIX_FILE_LOCKING`, `CONFIG_POSIX_FILE_SYSTEM_R`.

And then of course modify [the docs](https://docs.zephyrproject.org/latest/services/portability/posix/option_groups/index.html#posix-thread-safe-functions)

https://pubs.opengroup.org/onlinepubs/9799919799/xrat/V4_subprofiles.html

Reply:
All of the above should be in a separate PR.